ht16k33/src/main/java/com/google/android/things/contrib/driver/ht16k33/AlphanumericDisplay.java

```diff
@@ -102,23 +102,17 @@ public void display(String s) throws IOException {
         mBuffer.clear();
         mBuffer.mark();
         short n = (short) 0;
-        char prevChar = (char) 0;
+        char prevChar = '.';
         for (char c : s.toCharArray()) {
+            if (c == '.' && prevChar != '.') {
+                // add dot LED flag to the previous character.
+                n |= DOT;
+                mBuffer.reset();
+                mBuffer.putShort(n);
             // truncate string to the size of the display
-            if (mBuffer.position() == mBuffer.limit()) {
+            } else if (mBuffer.position() == mBuffer.limit()) {
```

Comment:
maybe we could simplify the tests if we only rely on the `position` being updated by the previous check:
```
 short prevFontData  = 0;
 for (char c : s.toCharArray()) {
   short fontData = (short) Font.DATA[c];
   if (c == '.') {
      mBuffer.reset();
      fontData = prevFontData | fontData;
   }
   if (mBuffer.position() == mBuffer.limit()) {
     break
   } 
   mBuffer.mark();
   mBuffer.putShort(fontData);
   prevFontData = fontData;
}
```